BayesDel.pm

```diff
@@ -0,0 +1,170 @@
+ https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA
+ Note: These files only contain pre-computed BayesDel scores for missense variants for assembly GRCh37.
+
+
+ For GRCh37:
+ tar zxvf BayesDel_170824_addAF.tgz
+ rm *.gz.tbi
+ gunzip *.gz
+ for f in BayesDel_170824_addAF_chr*; do grep -v "^#" $f >> BayesDel_170824_addAF.txt; done
+ cat BayesDel_170824_addAF.txt | sort -k1,1 -k2,2n > BayesDel_170824_addAF_sorted.txt
+ grep "^#" BayesDel_170824_addAF_chr1 > BayesDel_170824_addAF_all_scores.txt
+ cat BayesDel_170824_addAF_sorted.txt >> BayesDel_170824_addAF_all_scores.txt
+ bgzip BayesDel_170824_addAF_all_scores.txt
+ tabix -s 1 -b 2 -e 2 BayesDel_170824_addAF_all_scores.txt.gz
+
+ For GRCh38:
+ Remap GRCh37 file
+
+
+ The tabix utility must be installed in your path to use this plugin.
+
+=cut
+
+package BayesDel;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+
+sub new {
+  my $class = shift;
+
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+
+  my $param_hash = $self->params_to_hash();
+
+  # Test tabix
+  die "ERROR: tabix does not seem to be in your path\n" unless `which tabix 2>&1` =~ /tabix$/;
+
+  # Get file
+  if(defined($param_hash->{file})) {
+    die "ERROR: BayesDel file not not found!\n" unless -e $param_hash->{file};
+    $self->add_file($param_hash->{file});
+  }
+
+  # Get headers from file
+  my $headers;
+  open HEAD,"tabix -H $param_hash->{file} 2>&1 | ";
+  while(<HEAD>) {
+    chomp;
+    $_ =~ s/^\#//;
+    $headers = [split];
+  }
+  close HEAD;
+
+  die "ERROR: Could not read headers from $param_hash->{file}\n" unless defined($headers) && scalar @{$headers};
+
+  return $self;
+}
+
+sub feature_types {
+  return ['Feature', 'Intergenic'];
+}
+
+sub get_header_info {
+  return { BayesDel => 'Deleteriousness meta-score'};
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  my $vf = $tva->variation_feature;
+  my $transcript = $tva->transcript;
+
+  my $end = $vf->{end};
+  my $start = $vf->{start};
+  ($start, $end) = ($end, $start) if $start > $end;
+
+  my $ref_allele = $vf->ref_allele_string;
+  my $alt_allele = $tva->variation_feature_seq;
+  if ($vf->{strand} < 0) {
+    reverse_comp(\$ref_allele);
+    reverse_comp(\$alt_allele);
+  }
+
+  return {} unless $alt_allele =~ /^[ACGT]$/;
+
+  my ($res) = grep {
+    $_->{start} == $start &&
+    $_->{start} == $end &&
+    $_->{ref} eq $ref_allele &&
+    $_->{alt} eq $alt_allele
+  } @{$self->get_data($vf->{chr}, $start, $end)};
+
+  return defined $res->{result} ? $res->{result} : {};
+}
+
+sub parse_data {
+  my ($self, $line) = @_;
+  my ($chr, $pos, $ref, $alt, $score) = split /\t/, $line;
+
+  return {
+    start => $pos,
+    ref => $ref,
+    alt => $alt,
+    result => {
+      BayesDel => $score
+    }
+  };
+}
+
+sub get_start {
+  return $_[1]->{start};
```

Comment:
function get_end is missing

Reply:
The BayesDel file with pre-computed scores only has start.